cdk/domino_cdk/config/iam.py

```diff
@@ -0,0 +1,400 @@
+            ],
+            "Resource": [f"arn:aws:eks:*:{aws_account_id}:cluster/*"],
+        }
+    ]
+
+    if manual:
+        eks_nodegroups = []
+
+    cfn_tagging = {
+        "Effect": "Allow",
+        "Action": [
+            "ssm:AddTagsToResource",
+            "ssm:DeleteParameter",
+            "ssm:PutParameter",
+            "ssm:RemoveTagsFromResource",
+        ],
+        **from_cf_condition,
+        "Resource": [f"arn:aws:ssm:*:{aws_account_id}:parameter/CFN*"],
+    }
+
+    ssm = {
+        "Effect": "Allow",
+        "Action": ["ssm:GetParameters"],
+        "Resource": ["arn:aws:ssm:*::parameter/aws/service/eks/*"],
+    }
+
+    # TODO: TF-only version of this list
+    backup_plan = f"{stack_name}-efs"
+    if manual:
+        backup_plan = "*"
+
+    backup = {
+        "Effect": "Allow",
+        "Action": [
+            "backup:*BackupVault*",
+            "backup:CreateBackupPlan",
+            "backup:CreateBackupSelection",
+            "backup:DeleteBackupPlan",
+            "backup:DeleteBackupSelection",
+            "backup:GetBackupPlan",
+            "backup:GetBackupSelection",
+            "backup:ListTags",
+            "backup:TagResource",
+            "backup:UntagResource",
+        ],
+        **from_cf_condition,
+        "Resource": [
+            f"arn:aws:backup:*:{aws_account_id}:backup-vault:{stack_name}-efs",
+            f"arn:aws:backup:*:{aws_account_id}:backup-plan:{backup_plan}",
+        ],
+    }
+
+    if manual:
+        backup["Action"].remove("backup:CreateBackupPlan")
+        backup["Action"].remove("backup:DeleteBackupPlan")
+        backup["Action"].remove("backup:GetBackupPlan")
+        backup["Action"].insert(0, "backup:*BackupPlan")
+
+    backup_efs_kms_combo = [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "backup-storage:MountCapsule",
+                "elasticfilesystem:CreateAccessPoint",
+                "elasticfilesystem:CreateFileSystem",
+                "elasticfilesystem:DeleteAccessPoint",
+                "elasticfilesystem:DeleteFileSystem",
+                "elasticfilesystem:DescribeAccessPoints",
+                "elasticfilesystem:DescribeFileSystems",
+                "kms:CreateGrant",
+                "kms:Decrypt",
+                "kms:DescribeKey",
+                "kms:GenerateDataKey",
+                "kms:RetireGrant",
+            ],
+            **from_cf_condition,
+            "Resource": "*",
+        }
+    ]
+
+    if not manual:
+        backup_efs_kms_combo = []
+
+    ecr = [
+        {
+            "Effect": "Allow",
+            "Action": ["ecr:CreateRepository", "ecr:DeleteRepository"],
+            "Condition": {"ForAnyValue:StringEquals": {"aws:CalledVia": ["cloudformation.amazonaws.com"]}},
+            "Resource": [f"arn:aws:ecr:*:{aws_account_id}:repository/{stack_name}*"],
+        }
+    ]
+
+    if not manual:
+        ecr = []
+
+    if use_bastion:
+        bastion = [
```

Comment:
manual case heavily uses "*Resource" wildcards in part because otherwise the resulting policy with all GetResource, PutResource, DeleteResource, DescribeResources, ListResources, etc will not fit into default 6Kb policy size limit.

Since we have some additions here, I recommend to test with some long stack name and see if it still fits.

Or UT.

Reply:
I'll try a longer name for the moment and hope it still works, as if it works I'd like to just get it in for now. Whether it ends up being now or later, maybe the right thing to do is just split it into two policies. That's what we did for the old rancher-focused version of it.

Reply:
Your genius solution to split a policy into 2 policies and avoid 6K limit and use the same logic in both, is closing this.